### ISSUE TITLE
Bug: Favorites list empties unexpectedly.

### ISSUE THREAD

**Issue by laraws** [xiaoyifang/goldendict-ng#1837]
**Describe the bug**

Favorites list empties unexpectedly.

**OS and software versions**

Goldendict-ng 24.09.0.316ec90 at 2024-09-13T13:24:05Z
Qt 6.7.3 GCC 13.2.0 org.kde.Platform linux 6.8.0-47-generic x86_64-little_endian-lp64
Flags: MAKE_ZIM_SUPPORT MAKE_CHINESE_CONVERSION_SUPPORT NO_TTS_SUPPORT

- OS: ubuntu22.04
- Version flatpak 

I recently collected many words in my Favorites, but they've disappeared today and can't be recovered. I'm unsure why this happened. Has anyone else experienced this issue, or is there a bug in the app?

**Comment by shenlebantongying** [comment]
Not sure what happened, but the storage location of `favorite` is the same as the config files. Note that native and flatpak version uses different paths for config.

**Comment by shenlebantongying** [comment]
Not sure what happened, but the storage location of `favorite` is the same as the config files. Note that native and flatpak version uses different paths for config.

**Comment by laraws** [comment]
> Not sure what happened, but the storage location of `favorite` is the same as the config files. Note that native and flatpak version uses different paths for config.

Yeah. I know that the `favorite` file of flatpak is located in `~/.var/app/io.github.xiaoyifang.goldendict_ng/config/goldendict/`. I checked this directory's favorite file. It was empty, which is quite strange.

**Comment by laraws** [comment]
#1838

**Comment by laraws** [comment]
I get the answser. https://github.com/goldendict/goldendict/issues/1124

It's ok now.

**Comment by shenlebantongying** [comment]
I will double-check related code later. Data loss is not supposed to happen.
